### ISSUE TITLE
CLIP different text and image embedding sizes

### ISSUE THREAD

**Issue by asvskartheek** [lucidrains/DALLE-pytorch#4]
As per the CLIP paper's pseudo-code implementation, it has the feature to use different embedding sizes for image encoder and text encoder. To do this -
1. They have two additional linear layers - One for Text Encoded to Text embedded and another for images. 
2. These multimodal embeddings are L2 Normalized
3. The scaling factor used in the current implementation (dim**-0.5) is different from pseudo-code, which uses a learned temperature parameter.

Partial Pseudo-Code - 
```
I_f = image_encoder(I) #[n, d_i]
T_f = text_encoder(T) #[n, d_t]
# joint multimodal embedding [n, d_e]
I_e = l2_normalize(np.dot(I_f, W_i), axis=1)
T_e = l2_normalize(np.dot(T_f, W_t), axis=1)
# scaled pairwise cosine similarities [n, n]
logits = np.dot(I_e, T_e.T) * np.exp(t)
```

**Comment by lucidrains** [comment]
@asvskartheek Thank you! Let me know if this looks good https://github.com/lucidrains/DALLE-pytorch/commit/10ada06a4cf60de3446ec257078f8ef029f7547f

**Comment by asvskartheek** [comment]
I'm always amazed by your lightspeed implementations! This looks very much in line with the pseudo-code and closing the issue.
